core/src/main/java/org/mule/runtime/core/api/streaming/bytes/factory/AbstractCursorStreamProviderFactory.java

```diff
@@ -48,23 +49,45 @@ protected AbstractCursorStreamProviderFactory(ByteBufferManager bufferManager, S
     this.streamingManager = streamingManager;
   }
 
+  /**
+   * {@inheritDoc}
+   */
   @Override
-  public final Object of(EventContext eventContext, InputStream inputStream) {
+  public final Object of(EventContext eventContext, InputStream inputStream, ComponentLocation originatingLocation) {
     if (inputStream instanceof CursorStream) {
       return streamingManager.manage(((CursorStream) inputStream).getProvider(), eventContext);
     }
 
-    Object value = resolve(inputStream, eventContext);
+    Object value = resolve(inputStream, eventContext, originatingLocation);
     if (value instanceof CursorStreamProvider) {
       value = streamingManager.manage((CursorStreamProvider) value, eventContext);
     }
 
     return value;
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public final Object of(EventContext eventContext, InputStream inputStream) {
+    return of(eventContext, inputStream, eventContext.getOriginatingLocation());
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public final Object of(CoreEvent event, InputStream inputStream, ComponentLocation originatingLocation) {
+    return of(getRoot(event.getContext()), inputStream, originatingLocation);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
   @Override
   public final Object of(CoreEvent event, InputStream inputStream) {
-    return of(getRoot(event.getContext()), inputStream);
+    return of(getRoot(event.getContext()), inputStream, event.getContext().getOriginatingLocation());
```

Comment:
can you make the implementation null safe and pass no value instead of a wrong value for this case?

Reply:
Done.  It was already null safe (eg https://github.com/mulesoft/mule/blob/master/core-tests/src/test/java/org/mule/runtime/core/internal/streaming/AbstractTroubleshootCursorProviderTestCase.java#L66).  I changed those wrong values in all places.